jetty-core/jetty-websocket/jetty-websocket-core-server/src/main/java/org/eclipse/jetty/websocket/core/server/internal/CreatorNegotiator.java

```diff
@@ -59,7 +61,7 @@ public FrameHandler negotiate(ServerUpgradeRequest request, ServerUpgradeRespons
         }
         catch (Throwable t)
         {
-            callback.failed(t);
+            Response.writeError(request, response, callback, HttpStatus.INTERNAL_SERVER_ERROR_500, t.getMessage());
```

Comment:
I think `cb.failed()` is more correct, as it will fail a callback that may come from the upper layer (e.g. Servlet), so the right `Servlet` mechanism will be used to send an error.